### ISSUE TITLE
template doesn't work inside svg elements

### ISSUE THREAD

**Issue by bendavis78** [Polymer/polymer#1976]
For example:

``` html
    <dom-module id="test-element">
      <template>
        <svg class="clock" version="1.1" xmlns="http://www.w3.org/2000/svg">
          <template is="dom-repeat" items="{{positions}}" as="cx">
            <ellipse r="25" cx$="{{cx}}" cy="25" stroke="black"></ellipse> 
          </template>
        </svg>
      </template>
      <script>
        Polymer({
          is: 'test-element',
          ready: function() {
            this.positions = [0, 100, 200, 300];
          },
        });
      </script>
    </dom-module>
```

This produces the following JS error:

```
Uncaught TypeError: Cannot read property 'nodeType' of undefined
```

The problem seems to be that the dom template object doesn't have a content property. This worked fine in Polymer 0.5.

Demo here: http://jsbin.com/wiqowo/31/edit?html,output

**Comment by bendavis78** [comment]
It looks like some code didn't make it over from 0.5 that explicitly addressed this issue. Template elements with a `<svg>` parent have the same `namespaceURI` (`http://www.w3.org/2000/svg`), so the browser doesn't implement `HTMLTemplateElement`. The bootstrap code in 0.5 basically transplanted all child nodes of any svg-namespaced template into a new html-namespaced template, and then inserted that node back in its place.

If anyone else is looking for a workaround, you can put [this snippet](https://gist.github.com/bendavis78/15528ca2f501c44f2fa4) at the top of your element's `<script>` tag (it must be evaluated before call to `Polymer()`). It basically performs the same operation as it was done it 0.5.

**Comment by nomego** [comment]
We also hit this issue, thanks for the workaround @bendavis78 .

Will it be fixed in Polymer?

**Comment by canufeel** [comment]
@bendavis78 your snippet was a day saviour for me but it still has some problem. If you have more than one polymer element on the page and this svg element is not the first one the snippet breaks. This line:
    var root = doc.querySelector('dom-module > template').content;
should be changed so `dom-module` becomes the actual `#id-of-svg-element` which is put in the `<dom-module id="id-of-svg-element">`

**Comment by kevin-smets** [comment]
+1

**Comment by shmendo** [comment]
+1

**Comment by ocampesato** [comment]
Same error in Chrome 49, even with the tweaked version of svg-template-support.js, and also:
svg-template-support.js:4 Uncaught TypeError: Cannot read property 'content' of null

**Comment by raghav135** [comment]
+1 same error in latest chrome

**Comment by brian-cruickshank** [comment]
+1

**Comment by RossRothenstine** [comment]
+1

**Comment by christophe-g** [comment]
Thanks for the workaround - would be really nice to have templates within SVG work natively !

**Comment by joshuahiggins** [comment]
Does anyone know of a way to implement the workaround while still vulcanizing elements?

**Comment by christophe-g** [comment]
https://github.com/Polymer/polymer/pull/3372
C.

**Comment by brian-cruickshank** [comment]
Note that the [June 28, 2015 workaround](https://github.com/Polymer/polymer/issues/1976#issuecomment-116361820) does not work in Firefox 45.0.2 - it will actually prevent any of the svg from being rendered.

**Comment by boazhachlili** [comment]
Ran into this issue today. Any word on Polymer 1.0 support?

**Comment by andrew-silva-movile** [comment]
+1 same error

**Comment by fooloomanzii** [comment]
Although there is already a pull-request #3372 that could fix the namespace issue, here would be a work-around, that would build up the svg-node recursivly node by node.

There could be a container-element for svg:
```
<dom-module id="svg-container">
  <template>
    <svg id="svg" version="1.1" xmlns="http://www.w3.org/2000/svg"></svg>
      <content select="svg-component"></content>
  </template>
  <script>
    Polymer({
      is: 'svg-container',

      attached: function() {
        for (var i = 0; i < this.attributes.length; i++) {
          this.$.svg.setAttribute(Polymer.CaseMap.dashToCamelCase(this.attributes[i].name), this.attributes[i].value);
        }
      }
    });
  </script>
</dom-module>
```
and a element as svg-component:
```
<dom-module id="svg-component">
  <template>
    <content select="svg-component"></content>
  </template>
  <script>
    SvgComponent = Polymer({
      is: 'svg-component',

      _node: {},

      _namespace: 'http://www.w3.org/2000/svg',

      get rootElement () {
        if (this.parentNode.nodeName === 'SVG-CONTAINER')
          return this.parentNode.$.svg;
        else
          return this.parentNode._node;
      },

      factoryImpl: function(attributes) {
        for (var key in attributes)
          this.setAttribute(key, attributes[key]);
      },

      attached: function() {
        var is = this.attributes.is.value;

        if (is !== undefined) {
          this._node = document.createElementNS(this._namespace, is);
          for (var i = 0; i < this.attributes.length; i++) {
            if (this.attributes[i].name !== 'is')
              this._node.setAttribute(this.attributes[i].name, this.attributes[i].value);
          }
          this.rootElement.appendChild(this._node);
        }
      }
    });
  </script>
</dom-module>
```

Property binding would have to be formally and also event-binding, but you could use that to to do it automaticly. It would definitely be better, that SVG namespace is implemented in registration of new elements in Polymer.
 
And in a test: [plunker](http://plnkr.co/edit/pegYFTcr9koAZEg7pTMf?p=preview)

**Comment by Bodhiz** [comment]
@fooloomanzii Thanks a lot for the workaround. You saved me a lot of time ! 

+1 for native implementation.

**Comment by shahraship** [comment]
+1 for native implementation

**Comment by garryyao** [comment]
+1

**Comment by FlorianFe** [comment]
+1

**Comment by LukeusMaximus** [comment]
+1

**Comment by lao-tseu-is-alive** [comment]
+1 amazing to see one year later and still an issue

**Comment by AlexanderLang** [comment]
+1

**Comment by bcopy** [comment]
+1

**Comment by loekup** [comment]
@kevinpschaaf is this going to be possible in polymer 2 ?

**Comment by sz332** [comment]
This is still an issue :(

**Comment by garryyao** [comment]
Provided a workaround for now, tested in FF/Safari/IE, bundled/unbundled.
https://github.com/garryyao/polymer-svg-template

**Comment by alijaya** [comment]
any official workaround here?

**Comment by super-kamil** [comment]
+1

**Comment by jmonsma** [comment]
+1

**Comment by axax** [comment]
+1

**Comment by TrentBrown** [comment]
+1

**Comment by AbiSadhasivam** [comment]
+1

**Comment by renfeng** [comment]
+1

**Comment by renfeng** [comment]
For people looking for a working solution,
https://github.com/Polymer/polymer/issues/1976#issuecomment-225454535

**Comment by TimvdLippe** [comment]
We have a proposal for a core solution:

Detect if a `<template>` is inside an element that is in a SVG namespace (if it has `namespaceURI` which is SVG), then swap its contents into an `svg` element, do the parsing and then swap it back. This should be in `parse-template`.

The way to handle the transformation, is to expose a function `Polymer.svg` which transforms a tagged string into the corresponding svg namespace and returns the corrected template.

This whole process ensures that the elements are parsed in the correct namespace, to make sure the actual nodes are created. Then once the parsing has completed, we can swap the correctly constructed nodes back into the template.

We will be working on this issue this week to hopefully finally fix svg compatibility with templates + custom elements.

**Comment by TimvdLippe** [comment]
Above comment has been implemented in #5135. Would be great to get some feedback whether this is actually working for your use-cases :tada:

**Comment by kevinpschaaf** [comment]
Note full fix is blocked on https://github.com/webcomponents/shadycss/issues/166

**Comment by Neil65** [comment]
Is there a way forward with this issue? It seems like there is a Polymer solution (#5135) but it has been held back because of the possibility of addressing the issue at the platform level; however from what I can sense by reading https://github.com/whatwg/html/issues/3553 and https://github.com/whatwg/html/issues/3563 it does not seem that a platform level solution is likely any time soon.
